sidechain/fork-tree/src/lib.rs

```diff
@@ -0,0 +1,1547 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Utility library for managing tree-like ordered data with logic for pruning
+//! the tree while finalizing nodes.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![warn(missing_docs)]
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+#[macro_use]
+extern crate sgx_tstd as std;
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use std::vec::Vec;
```

Comment:
[Optional]
I think we could do this:
```suggestion
use alloc::vec::Vec;
```

Not that much of a benefit here as `std` will be needed for other things and rewriting those would mean much difference compared to the original file, but still one less `std` usage :).

----

Just a thought: let's say we would change the `std::` usages to `core/alloc`, do you think they would be interested in accepting a patch like that? I quickly checked all  `std::` stuff and all of them come from `core` aside from `Vec`. 
Would it make sense to do this, or just go with `sgx_tstd` and call it a day? @clangenb @coax1d What is your take her?

Reply:
Ah this is actually good input. I should have waited for your review. :) I mean in general, if we can live with core only, it definitely makes sense to remove the `sgx_tstd` dependency. :) Will gladly accept the PR.

Reply:
Sorry, what I meant was that we should submit the PR to paritytech/substrate so there won't be that much of a difference between our fork and upstream. :)

Reply:
Yeah, this ok too. :) However, I am curious. I think I quickly checked it and thought that some imports were not available in core. If there is no issue, you can gladly supply the PR parity.

Lastly, @coax1d are you going to make some amendments to the fork-tree in your upcoming PRs? If yes, we need to have the local copy anyhow.